### PR TITLE
Create node.js.yml for github actions

### PR DESCRIPTION
**Pull request by riverole**
It will just run npm test after a push to master or a pull request against master.

### CODE REVIEW

**Comment by riverole** [comment]
Ouch... of course it doesn't work, because there's no api keys... :S. I don't know how we can overcome this though...

**Comment by riverole** [comment]
I guess we might be able to use this: https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets, so we can use some test keys for running these tests.

**Comment by riverole** [comment]
I don't seem to have the ability to create secrets.

**Comment by riverole** [comment]
Ok, so I was testing a bit by forking the repository (so I was able to create the secrets). I think I got it. I will be sending a CR tomorrow probably, and you just need to create the secrets with specific names (with the api keys, maybe a second set of them) and then this should work.

**Comment by riverole** [comment]
So, basically, this seems to be ready to go. However, the tests are kind of broken right now, so they're failing. I have them fixed in my copying-resources pull request, so I wouldn't necessarily worry about that for the moment. It's kind of surprising that this works, so I guess you already created the secrets with the client and the picker @lorennorman ?

**Comment by riverole** [comment]
Also, last question, do you want it to run in both 12.x and 14.x? They run in parallel, so it's not like it takes more time, but I guess there's no really need to run in both. What are you guys' thoughts?

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x]
```
Comment (lorennorman) [inline .github/workflows/node.js.yml:19]: Just the latest is fine.

File: .github/actions/copyAPIKeys/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh -l
+
+JSON_FMT='{"google": {"client_id": "%s", "picker": "%s", "fonts": "%s"}}\n'
+
+printf "$JSON_FMT" "$CLIENT_ID" "$PICKER" "$PICKER" > .api_keys.dev.json
```
Comment (lorennorman) [inline .github/actions/copyAPIKeys/entrypoint.sh:5]: I'm thinking just copy the example file here, the keys are never actually used, but the file needs to be present.
`cp example.api_keys.dev.json api_keys.dev.json`

Does that simplify other things?

**Comment by lorennorman** [comment]
This is amazing, great work!

No, I have not put any secrets in, I see that the tests are running though! The tests themselves shouldn't need the keys, but I bet it was crashing before due to there not being a key file to load. Since you added scripts that write things into the key file, it isn't throwing an error anymore.

I say we just run on the latest Ubuntu. I wonder how the testing resources are shared/billing/rationed...

Another thing, I think my test script violently slams shut (second run says "cancelled") when one of the two programs (http server and cypress) quits, I'm thinking we'll have to fix that too when we get the tests green.

This is awesome!

**Comment by riverole** [comment]
Ohhh! that explains it! I will change it then, I think I can even get rid of the action, and just do a bash command there in the node file.

**Comment by riverole** [comment]
Created #67 for cleaning up, since now it is way simpler, so didn't want to pollute the commit history.

**Pull request closed without merging**